src/attestation/Attester.chain.ts

```diff
@@ -42,23 +47,24 @@ export default class AttesterChain extends Attester implements IAttesterChain {
   /**
    * Generates a new key pair and returns a new [[AttesterChain]].
    *
-   * @param validityDuration The duration in days for which the public key will be valid.
-   * @param maxAttributes The maximum number of attributes that can be signed with the generated private key.
-   * @param keypairType The signature scheme used in the keyring pair, either 'sr25519' or 'ed25519'.
+   * @param options An optional object containing options for the key generation.
+   * @param options.validityDuration The duration in days for which the public key will be valid.
+   * @param options.maxAttributes The maximum number of attributes that can be signed with the generated private key.
+   * @param options.keyLength The key length of the new key pair. Note that this key will only support credentials and claimer with the same key length.
+   * @param options.keypairType The signature scheme used in the keyring pair, either 'sr25519' or 'ed25519'.
    * @returns A [[AttesterChain]] instance including a chain address and a public and private key pair.
    */
   public static async create(
-    validityDuration?: number,
-    maxAttributes = 70,
-    keypairType: KeypairType = 'sr25519'
+    options: ChainKeyGenOptions = {}
   ): Promise<AttesterChain> {
-    const durationInNanoSecs = daysToNanoSecs(validityDuration || 365)
```

Comment:
this was a bug. We also convert the duration in `genKeyPair` the duration would be converted twice from days to ns.